packages/near-sdk-js/src/utils.ts

```diff
@@ -147,6 +197,105 @@ export function deserialize(valueToDeserialize: Uint8Array): unknown {
+      if (ty !== undefined && ty.hasOwnProperty("map")) {
+        for (const mkey in value) {
+          if (ty["map"]["value"] === "string") {
+            class_instance[key][mkey] = value[mkey];
+          } else {
+            class_instance[key][mkey] = decodeObj2class(
+              new ty["map"]["value"](),
+              value[mkey]
+            );
+          }
+        }
+        // eslint-disable-next-line no-prototype-builtins
+      } else if (ty !== undefined && ty.hasOwnProperty("array")) {
+        for (const k in value) {
+          if (ty["array"]["value"] === "string") {
+            class_instance[key].push(value[k]);
+          } else {
+            class_instance[key].push(
+              decodeObj2class(new ty["array"]["value"](), value[k])
+            );
+          }
+        }
+        // eslint-disable-next-line no-prototype-builtins
+      } else if (ty !== undefined && ty.hasOwnProperty(UNORDERED_MAP_SCHE)) {
+        class_instance[key]._keys.length = obj[key]._keys.length;
+        class_instance[key].constructor.schema = ty;
+        const subtype_value = ty[UNORDERED_MAP_SCHE]["value"];
+        class_instance[key].subtype = function () {
+          return subtype_value;
+        };
+        // eslint-disable-next-line no-prototype-builtins
+      } else if (ty !== undefined && ty.hasOwnProperty(VECTOR_SCHE)) {
+        class_instance[key].length = obj[key].length;
+        class_instance[key].constructor.schema = ty;
+        const subtype_value = ty[VECTOR_SCHE]["value"];
+        class_instance[key].subtype = function () {
+          return subtype_value;
+        };
+        // eslint-disable-next-line no-prototype-builtins
+      } else if (ty !== undefined && ty.hasOwnProperty(UNORDERED_SET_SCHE)) {
+        class_instance[key]._elements.length = obj[key]._elements.length;
+        class_instance[key].constructor.schema = ty;
+        const subtype_value = ty[UNORDERED_SET_SCHE]["value"];
+        class_instance[key].subtype = function () {
+          return subtype_value;
+        };
+        // eslint-disable-next-line no-prototype-builtins
+      } else if (ty !== undefined && ty.hasOwnProperty(LOOKUP_MAP_SCHE)) {
+        class_instance[key].constructor.schema = ty;
+        const subtype_value = ty[LOOKUP_MAP_SCHE]["value"];
+        class_instance[key].subtype = function () {
+          return subtype_value;
+        };
+        // eslint-disable-next-line no-prototype-builtins
+      } else if (ty !== undefined && ty.hasOwnProperty(LOOKUP_SET_SCHE)) {
+        class_instance[key].constructor.schema = ty;
+        const subtype_value = ty[LOOKUP_SET_SCHE]["value"];
+        class_instance[key].subtype = function () {
+          return subtype_value;
+        };
+      } else {
+        // normal class
+        class_instance[key].constructor.schema =
+          class_instance.constructor.schema[key];
+        class_instance[key] = decodeObj2class(class_instance[key], obj[key]);
+      }
```

Comment:
Can you also test a case, where a custom class contains a UnorderedMap:

```
Class Truck {
    static schema = {
        name: "string",
        speed: "number",
        load: {unordered_map: {value: 'string'}}
    };
// ...
}
```

Reply:
ok, I will add it

Reply:
added!

Reply:
Great!